mbed_greentea/mbed_greentea_cli.py

```diff
@@ -250,6 +299,18 @@ def main():
                     action="store_true",
                     help='List available binaries')
 
+    parser.add_option('', '--list-test-cases',
+                    dest='list_test_cases',
+                    default=False,
+                    action="store_true",
+                    help='List available test cases inside the binaries')
+
+    parser.add_option('', '--list-test-plan',
+                    dest='list_test_plan',
+                    default=False,
+                    action="store_true",
+                    help='List implemented, not implemented, and conflicting test cases from the specified Test Plan (--test-plan)')
+
```

Comment:
Perhaps this should be mentioned in the PR. This PR is destined for ```testlink``` branch. I intend to keep this branch in sync with master and release separate versions.
The reason of having separate switches is to maintain compatibility with old behaviour. Once we migrate to using features implemented here we can then deprecate  and converge the switches.